lib/rucio/core/replica.py

```diff
@@ -1431,12 +1493,18 @@ def __bulk_add_file_dids(files, account, dataset_meta=None, *, session: "Session
     for f in files:
         condition.append(and_(models.DataIdentifier.scope == f['scope'], models.DataIdentifier.name == f['name'], models.DataIdentifier.did_type == DIDType.FILE))
 
-    q = session.query(models.DataIdentifier.scope,
-                      models.DataIdentifier.name,
-                      models.DataIdentifier.bytes,
-                      models.DataIdentifier.adler32,
-                      models.DataIdentifier.md5).with_hint(models.DataIdentifier, "INDEX(dids DIDS_PK)", 'oracle').filter(or_(*condition))
-    available_files = [dict([(column, getattr(row, column)) for column in row._fields]) for row in q]
+    stmt = select(
+        models.DataIdentifier.scope,
+        models.DataIdentifier.name,
+        models.DataIdentifier.bytes,
+        models.DataIdentifier.md5,
+        models.DataIdentifier.adler32,
+    ).with_hint(
+        models.DataIdentifier, 'INDEX(DIDS DIDS_PK)', 'oracle'
+    ).where(
+        or_(*condition)
+    )
+    available_files = [dict([(column, getattr(row, column)) for column in row._fields]) for row in session.execute(stmt)]
```

Comment:
This is a pretty nasty definition. Can I recommend: 
```
query = session.execute(stmt).scalars()
available_files  = [
   {column: row_attr for column, row_attr in row.items()} 
    for row 
    in query
]
```

Reply:
What do you think of using the ._asdict() method here? I changed this to:

```
    available_files = [res._asdict() for res in session.execute(stmt).all()]
```

Reply:
If it passes the tests, I see no reason why not! _asdict doesn't seem to be private here according to the sqla docs. I'll resolve all my other comments you solved this way.